integtests/components/test_tools.py

```diff
@@ -79,3 +85,155 @@ async def test_retrieve_components_by_types(mcp_context: Context, configs: list[
     for item in result:
         assert isinstance(item, ComponentWithConfigurations)
         assert item.component.component_type == 'extractor'
+
+
+@pytest.mark.asyncio
+async def test_create_component_root_configuration(mcp_context: Context, configs: list[ConfigDef]):
+    """Tests that `create_component_root_configuration` creates a configuration with correct metadata."""
+
+    # Use the first component from configs for testing
+    test_config = configs[0]
+    component_id = test_config.component_id
+
+    test_name = 'Test Configuration'
+    test_description = 'Test configuration created by automated test'
+    test_parameters = {}
+    test_storage = {}
+
+    # Create the configuration
+    result = await create_component_root_configuration(
+        ctx=mcp_context,
+        name=test_name,
+        description=test_description,
+        component_id=component_id,
+        parameters=test_parameters,
+        storage=test_storage
+    )
+    try:
+        assert isinstance(result, ComponentRootConfiguration)
+        assert result.configuration_name == test_name
+        assert result.configuration_description == test_description
+        assert result.component_id == component_id
+        assert result.configuration_id is not None
+        assert result.parameters == test_parameters
+        assert result.storage == test_storage
+
+        # Verify the configuration exists in the backend by fetching it
+        client = KeboolaClient.from_state(mcp_context.session.state)
+        config_detail = await client.storage_client.configuration_detail(
+            component_id=component_id,
+            configuration_id=result.configuration_id
+        )
+
+        assert config_detail['name'] == test_name
+        assert config_detail['description'] == test_description
+        assert 'configuration' in config_detail
+
+        # Verify the metadata - check that KBC.MCP.createdBy is set to 'true'
+        metadata = await client.storage_client.configuration_metadata_get(
+            component_id=component_id,
+            configuration_id=result.configuration_id
+        )
+
+        # Convert metadata list to dictionary for easier checking
+        # metadata is a list of dicts with 'key' and 'value' keys
+        assert isinstance(metadata, list)
+        metadata_dict = {item['key']: item['value'] for item in metadata if isinstance(item, dict)}
+        assert 'KBC.MCP.createdBy' in metadata_dict
+        assert metadata_dict['KBC.MCP.createdBy'] == 'true'
+    finally:
+        # Delete the configuration
+        await client.storage_client.configuration_delete(
+            component_id=component_id,
+            configuration_id=result.configuration_id
+        )
```

Comment:
please call the configuration_delete twice - to delete the configuration from the recycle bin as well